jwst/extract_2d/extract_2d.py

```diff
@@ -13,18 +13,24 @@
 log.setLevel(logging.DEBUG)
 
 
-def extract2d(input_model, which_subarray=None, apply_wavecorr=False, reffile="", grism_objects=[]):
+def extract2d(input_model, which_subarray=None, apply_wavecorr=False, reference_files={}, grism_objects=[]):
+
+    if not any(reference_files.values()):
+        log.critical("extract2d needs reference files to make extractions, none were passed")
```

Comment:
Is this true (that extract_2d needs reference files) for all cases in which extract_2d is used, including, for example, NIRSpec fixed-slit and MOS? What reference files does it use?

Reply:
grisms always needs the wavelengthrange reference file.  nirspec is using the wavecorr file for NRS_FIXEDSLIT, NRS_MSASPEC, NRS_BRIGHTOBJ. If we want to avoid doing checks we need to move the exp_type checking logic up to the step level

Reply:
okay - just removing the warning, that should make NRS_LAMP go through with no reference files and the other classes can check reference files for themselves :)

Reply:
👍  Just didn't realize that all the modes needed some kind of ref file. Now I know.